cmd/statusd/main.go

```diff
@@ -158,6 +168,28 @@ func startCollectingStats(interruptCh <-chan struct{}, nodeManager common.NodeMa
 	}
 }
 
+func syncAndStopNode(nodeManager common.NodeManager) (err error) {
+	log.Println("Node will synchronize and exit")
+	v := *syncAndExit
+	if v < 0 {
+		return errors.New("negative timeout value")
+	}
+	if v == 0 {
+		err = nodeManager.EnsureSync(context.Background())
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), (time.Duration)(v)*time.Minute)
+		err = nodeManager.EnsureSync(ctx)
+		defer cancel()
+	}
+	if err != nil {
+		return
+	}
+	var done <-chan struct{}
+	done, err = nodeManager.StopNode()
```

Comment:
```go
done, err := nodeManager.StopNode()
<-done
``` 
should work just fine.

Technically, it should also have a timeout but I think we can skip it.

Reply:
It returns `(err error)` so `return` is compulsory there. And I wanted it to write the error to existing var (return var).